src/server/dcp_server/models.py

```diff
@@ -60,12 +75,475 @@ def masks_to_outlines(self, mask):
         """        
         return utils.masks_to_outlines(mask) #[True, False] outputs
 
+class CellFullyConvClassifier(nn.Module):
+    
+    '''
+    Fully convolutional classifier for cell images.
+
+    Args:
+        in_channels (int): Number of input channels.
+        num_classes (int): Number of output classes.
+    '''
+
+    def __init__(self, model_config, train_config, eval_config):
+        super().__init__()
+
+        self.in_channels = model_config.get("in_channels", 1)
+        self.num_classes = model_config.get("num_classes", 3)
+
+        self.train_config = train_config
+        self.eval_config = eval_config
+
+        self.layer1 = nn.Sequential(
+            nn.Conv2d(self.in_channels, 16, 3, 2, 5),
+            nn.BatchNorm2d(16),
+            nn.ReLU(),
+            nn.Dropout2d(p=0.2),
+        )
+
+        self.layer2 = nn.Sequential(
+            nn.Conv2d(16, 64, 3, 1, 1),
+            nn.BatchNorm2d(64),
+            nn.ReLU(),
+            nn.Dropout2d(p=0.2),
+        )
+
+        self.layer3 = nn.Sequential(
+            nn.Conv2d(64, 128, 3, 2, 4),
+            nn.BatchNorm2d(128),
+            nn.ReLU(),
+            nn.Dropout2d(p=0.2),
+        )
+
+        self.final_conv = nn.Conv2d(128, self.num_classes, 1)
```

Comment:
should this be a fully connected layer rather than a convolution?

Reply:
We use patches of the same size because we pad them, but we can also use it for patches of different sizes with batch_size=1 and lower learning rate. Additionally, I still think fully convolutional network is more flexible and has fewer parameters.

Reply:
okay thanks for the explanation